tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/ConstSimplifierBase.scala

```diff
@@ -29,49 +29,170 @@ abstract class ConstSimplifierBase {
 
     // integer operations
     case OperEx(TlaArithOper.plus, ValEx(TlaInt(left)), ValEx(TlaInt(right))) =>
+      ValEx(TlaInt(left + right))(intTag)
+
+    // 0 + x = x
+    case OperEx(TlaArithOper.plus, ValEx(TlaInt(left)), rightEx) =>
       if (left == 0) {
-        ValEx(TlaInt(right))(intTag)
-      } else if (right == 0) {
-        ValEx(TlaInt(left))(intTag)
+        rightEx
+      } else {
+        OperEx(TlaArithOper.plus, ValEx(TlaInt(left))(intTag), rightEx)(intTag)
+      }
+
+    // x + 0 = x
+    case OperEx(TlaArithOper.plus, leftEx, ValEx(TlaInt(right))) =>
+      if (right == 0) {
+        leftEx
       } else {
-        ValEx(TlaInt(left + right))(intTag)
+        OperEx(TlaArithOper.plus, leftEx, ValEx(TlaInt(right))(intTag))(intTag)
       }
```

Comment:
Don't mean to be code golfing here, and I know it's a bit orthogonol to these changes, but do you think a set of patterns like this might work here? 

```scala
    case OperEx(TlaArithOper.plus, ValEx(TlaInt(0)), ex) => ex
    case OperEx(TlaArithOper.plus, ex, ValEx(TlaInt(0))) => ex
    case ex @ OperEx(TlaArithOper.plus, _, _) => ex
```

If so, I think it's reads much more clearly. If would work, and you agree it's easier to read, you might consider swapping that out here and in the other similar cases.

Reply:
I'll +1 this sentiment, I feel like there's some pattern-match simplification that can happen here. Either flat, like Shon suggested or something of the sort:
```
case OperEx( arithOper, l, r ) =>
  (l,r) match {
   case (literal, literal) =>
   case (any, literal) => ...
   ...
  }
```

Reply:
I've just comitted a version similiar to the one Shon proposed. I couldn't actually do `ValEx(TlaInt(0))` because 0 has to be a `BigInt`, and I can't do `ValEx(TlaInt(BigInt(0)))` because `BigInt` is not a case class. I could do something like this: https://codereview.stackexchange.com/questions/14561/matching-bigints-in-scala, but I found it more idiomatic to instead use actual if expressions inside the case statements like
```scala
case OperEx(TlaArithOper.plus, ValEx(TlaInt(left)), ex) if left == 0 => ex
```

This helped a lot with readability and I've also grouped this flat simplifications by arithmetical operation, which I personally believe is better to reason about than having a generalized form that avoids code duplication and increases complexity. But that's my point of view, let me know what you guys think :smile:

Reply:
Sorry I missed this! Makes perfect sense to me. TIL :)